tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/AnalyzerTests.cs

```diff
@@ -178,62 +193,62 @@ public async Task NegativeTest()
         }
 
         [InlineData("UA0001")]
-        [InlineData("UA0002")]
-        [InlineData("UA0003")]
-        [InlineData("UA0004")]
+        [InlineData("HtmlStringUpgrade")]
+        [InlineData("ResultUpgrade")]
+        [InlineData("FilterUpgrade")]
         [InlineData("UA0005")]
         [InlineData("UA0006")]
         [InlineData("UA0007")]
         [InlineData("UA0008")]
-        [InlineData("UA0009")]
+        [InlineData("HelperResultUpgrade")]
         [InlineData("UA0010")]
         [InlineData("UA0012")]
-        [InlineData("UA0013")]
+        [InlineData("ControllerUpgrade")]
         [Theory]
-        public async Task UpgradeAnalyzers(string diagnosticId)
+        public async Task UpgradeAnalyzers(string scenarioName)
         {
             foreach (var language in new[] { Language.CSharp, Language.VisualBasic })
             {
-                var expectedDiagnostics = ExpectedDiagnostics[diagnosticId].Where(diagnostics => diagnostics.Language == language);
+                var expectedDiagnostics = ExpectedDiagnostics[scenarioName].Where(diagnostics => diagnostics.Language == language);
                 if (!expectedDiagnostics.Any())
                 {
                     // nothing to see here, move along
                     continue;
                 }
 
                 var fileExtension = language.GetFileExtension();
-                var diagnostics = await TestHelper.GetDiagnosticsAsync(language, $"{diagnosticId}.{fileExtension}", diagnosticId).ConfigureAwait(false);
+                var diagnostics = await TestHelper.GetDiagnosticsAsync(language, $"{scenarioName}.{fileExtension}", expectedDiagnostics.Select(e => e.Id).Distinct()).ConfigureAwait(false);
                 AssertDiagnosticsCorrect(diagnostics, expectedDiagnostics);
             }
         }
 
         [InlineData("UA0001")]
-        [InlineData("UA0002")]
-        [InlineData("UA0003")]
-        [InlineData("UA0004")]
+        [InlineData("HtmlStringUpgrade")]
```

Comment:
Do we need tests for each of these? Going forward, are you expecting a full code analysis test for each upgrade pair? What about verifying the list contains something specific?

Reply:
I think we may as well leave the tests since we already have them. I like the idea of having a test to make sure that expected pairs are present, though. I'll add that and, in the future, we can just update the unit test when new pairs are added.